### PR TITLE
Improve test coverage related to 'Rect'

### LINKED ISSUE #26: 테스트 코드 보강하기
라이브러리가 잘 동작하는지 확인하기 위해 테스트 코드를 사용합니다. 테스트 코드는 견고할수록 좋습니다. 현재 작성되어 있는 테스트 코드들을 확인하고 미처 확인하지 못한 동작이 있다면 추가해 봅시다.

### PR DESCRIPTION
**Pull request by ChrisLee02**
## What

Strengthened `bracket-geometry::Rect` test coverage.

- Added a default-value test to confirm `Rect::default()` matches `Rect::zero()`
- Expanded `intersect` coverage for contained, partial-overlap, identical, edge-touching, vertex-touching, and disjoint cases
- Expanded `center` coverage for even-sized, odd-sized, and negative-origin rectangles
- Expanded `point_in_rect` coverage for interior, edge, exclusive-boundary, and outside points
- Updated `for_each` validation to assert deterministic iteration order

## Why

`Rect` is a foundational geometry type, and its boundary behavior needs explicit test coverage to prevent regressions.

These additions document the intended semantics around intersection, center calculation, point inclusion, and iteration order, especially for edge cases that are easy to misinterpret or break during refactors.

Closes #26.

### Required

- [x] `cargo check --all` passes
- [x] `cargo fmt --all -- --check` passes
- [x] `cargo clippy --workspace --all-targets -- -D warnings -A clippy::multiple-crate-versions` passes
- [x] `cargo test --all` passes
- [x] I linked the related issue (for example: `Closes #123`)

### Functional Validation

- [ ] Behavior related to this change was verified locally (if applicable)
- [ ] Rendering/backend behavior was verified when runtime code changed (if applicable)
- [ ] Algorithm behavior (pathfinding/FOV/noise/random) was verified when affected (if applicable)
- [ ] I added or updated tests for changed behavior (if applicable)

### Configuration & Docs

- [ ] User-facing docs were updated (`README.md`, `ARCHITECTURE.md`, or relevant manual pages, if applicable)
- [ ] New dependencies/configuration are documented (if applicable)
- [ ] No sensitive values or credentials were introduced

### If Applicable

- [ ] Security impact considered (run `cargo audit` locally if needed)
- [ ] Breaking behavior changes are clearly described in this PR

## Summary by CodeRabbit

* **Tests**
  * Added checks ensuring default rectangle state equals the zero rectangle.
  * Expanded intersection tests to cover containment, partial overlap, identical, edge-touching, corner-touching, and non-overlapping scenarios.
  * Strengthened center calculations by asserting full-point equality.
  * Verified ordered traversal of rectangle points by collecting and comparing the exact sequence.

[Review Change Stack](https://app.coderabbit.ai/change-stack/utilForever/roguekit/pull/67)

### CODE REVIEW

File: bracket-geometry/src/rect.rs [hunk 1]
```diff
@@ -167,24 +174,55 @@ mod tests {
...
 
     #[test]
     fn test_center() {
         let r1 = Rect::with_size(0, 0, 10, 10);
-        let center = r1.center();
-        assert!(center.x == 5 && center.y == 5);
+        assert_eq!(r1.center(), Point::new(5, 5));
+
+        let r2 = Rect::with_size(0, 0, 11, 11);
+        assert_eq!(r2.center(), Point::new(5, 5));
```
Comment (utilForever) [inline bracket-geometry/src/rect.rs:209]: 정수 타입으로 나타내야 하다 보니 `(0, 0)`과 `(11, 11)`의 중앙이 `(5, 5)`가 되는게 맞긴 한데, 이렇게 구하는 게 맞는지 좀 생각해 봅시다.

File: bracket-geometry/src/rect.rs [hunk 2]
```diff
@@ -167,24 +174,55 @@ mod tests {
...
     }
 
     #[test]
     fn test_point_in_rect() {
         let r1 = Rect::with_size(0, 0, 10, 10);
+        // case 1: point is strictly inside the rectangle
         assert!(r1.point_in_rect(Point::new(5, 5)));
+        // case 2: point is on the edge
+        assert!(r1.point_in_rect(Point::new(0, 5)));
+        // case 3: point is in the rect only if x < x2 and y < y2
```
Comment (utilForever) [inline bracket-geometry/src/rect.rs]:
```suggestion
        // case 3: point is in the rect only if x1 < x2 and y1 < y2
```

File: bracket-geometry/src/rect.rs [hunk 3]
```diff
@@ -200,16 +238,23 @@ mod tests {
 
     #[test]
     fn test_rect_callback() {
```
Comment (utilForever) [inline bracket-geometry/src/rect.rs:240]: 저는 `Vec`에 넣는 방식이 별로라고 생각합니다. 그 이유는 `HashMap`을 사용하면 순서에 의존받지 않지만, `Vec`을 사용하면 순서에 의존받기 때문입니다. 예를 들어, `for_each`의 내부 동작이 y → x 순서에서 x → y 순서로 바뀌었다고 가정해 봅시다. 그러면 저장되는 값들은 달라지지 않지만 그 순서는 달라질 수 있습니다.

**Review by utilForever — CHANGES_REQUESTED** [review]
Can you check my comment? Thanks.

File: bracket-geometry/src/rect.rs (hunk 1 above)
Comment (ChrisLee02) [reply bracket-geometry/src/rect.rs:209]: 테스트 코드의 보강이 PR의 목적인 만큼, Rect에 대한 동작의 현재 구현 계약을 테스트 코드에 명시하는 것이 적절할 것 같습니다. 말씀주신 부분은 후속 이슈로 다뤄볼 만 할 것 같아요

File: bracket-geometry/src/rect.rs (hunk 3 above)
Comment (ChrisLee02) [reply bracket-geometry/src/rect.rs:240]: for_each 내부 동작이 바뀌는 경우에 테스트 코드도 함께 맞춰주는게 더 적절하지 않을까요?

Comment (ChrisLee02) [reply bracket-geometry/src/rect.rs:240]: traverse order가 중요하지 않다면 / 혹은 중요하다면
각각이 코멘트에 확실히 들어가는게 첫 번째로 필요할 것 같습니다.

그런 코멘트가 존재하지 않는 상황이라면 테스트 코드의 역할은 구현의 observable behavior를 최대한 촘촘히 검증하는 것이니까요.

File: bracket-geometry/src/rect.rs (hunk 2 above)
Comment (ChrisLee02) [reply bracket-geometry/src/rect.rs]: https://github.com/utilForever/roguekit/pull/67/commits/d56640ca29882eb631fa0458f9cc02310c7773d5

**Comment by ChrisLee02** [comment]
https://github.com/utilForever/roguekit/pull/67/files#diff-bfacd6b38b5fe501147b906d2b7a93dd9044a135df40d9df50df833174a49c43

반영해서 1개 커밋만 force-push했습니다.

File: bracket-geometry/src/rect.rs (hunk 2 above)
Comment (utilForever) [reply bracket-geometry/src/rect.rs]: 수정 확인했습니다.

File: bracket-geometry/src/rect.rs (hunk 3 above)
Comment (utilForever) [reply bracket-geometry/src/rect.rs:240]: `Rect` 구조체에서 점이 어떤 순서로 저장될 지는 내부 구현에 따라 달라질 수 있습니다. 예를 들어, 어떤 점을 맨 처음에 저장할 지도 다를 수 있고, 시계 방향 순서로 저장할 지 / 반시계 방향 순서로 저장할 지도 다를 수 있습니다. 그래서 어떤 점이 들어있는지 확인할 때 순서는 배제하는 게 좋다는 게 저의 의견입니다.

만약 코멘트가 필요하다면 이 부분을 추가하는 게 좋겠습니다. 혹시 다른 의견이 있으시다면 남겨주세요.

Comment (ChrisLee02) [reply bracket-geometry/src/rect.rs:240]: Rect 구조체에서 점이 어떤 순서로 순회될지가 코드 상에 고정되어 있는 것으로 이해했는데요.
순회 순서가 바뀐다고 하면 그건 rect.rs 코드가 변경되는 상황일테고 그때 변경한 사람이 테스트까지 함께 수정해주는게 더 적절하지 않을까요?

Comment (utilForever) [reply bracket-geometry/src/rect.rs:240]: 현재 코드가 미래에 어떻게 바뀔지는 모릅니다. 테스트 코드에 담아야 하는 건 우리가 이 함수를 호출했을 때 기대하는 결과값이 의도와 맞는가라고 생각하는데, 이 테스트에서 기대하는 건 무엇일까요?

1. 새 `Rect`를 만들었을 때 이 점들이 들어 있어야 한다.
2. 새 `Rect`를 만들었을 때 이 점들이 이 순서로 들어 있어야 한다.

1번이 의도라면 순서는 배제되어야 하고, 2번이 의도라면 순서가 고려되어야 합니다. 무엇을 의도했는지 생각해 보시고, 의도에 맞춰서 진행하도록 하죠.

Comment (ChrisLee02) [reply bracket-geometry/src/rect.rs:240]: 1. 현재 코드가 미래에 바뀐다면, 그때 테스트가 깨져야 변경자가 확인한 후 기존 코드의 계약을 확인하고 침습 범위를 확인해서 예상치 못한 behavior change를 방지할 수 있을 것입니다.
2. for_each가 pure function이라면 결과값 검증으로 충분하겠지만, 부수효과를 동반한다면 세부 동작을 테스트에 더 자세히 고정해야 합니다.

말씀주신 기댓값 검증은 test_rect_set에서 수행 중인 것 같고, test_rect_callback는 미래 코드 변경에 대해 부수효과로 인한 문제를 방지하기 위한 안전장치에 더 가까워보이네요.

Comment (utilForever) [reply bracket-geometry/src/rect.rs:240]: 1. 제가 과거에 작성했던 다른 프로젝트들의 코드를 좀 살펴보고 왔는데, `Rect` 구조체가 잘못 설계되어 있다는 생각이 듭니다. 현재는 각 필드에 직접 접근이 가능해서 사각형의 형태가 보장 안되게 만들 수 있습니다. 이 부분은 향후 다른 이슈를 통해서 개선하는 게 좋겠습니다.
2. 1번을 작업할 때 `for_each`는 삭제하는 게 좋겠습니다. `Rect`에서 수행할 수 있는 동작들을 메소드로 구현하면 굳이 있어야 할 필요가 없을 것입니다.
3. 현재 PR 기준으로 논의하자면, 개인적으로는 순서가 중요하지 않다고 생각하지만 `for_each` 함수가 존재한다면 이 함수를 사용하는 사용자가 향후 다른 형태로 변환해서 다시 사용할 가능성이 존재하고 이 중 순서가 중요한 동작도 있을 수 있습니다. 종합적으로 고려할 때, 수정한 코드를 유지하는게 좋겠다는 결론에 이르렀습니다.

따라서 이 수정 사항을 유지하고, `Rect` 구조를 다시 만드는 작업을 이슈로 등록하고자 합니다.

**Review by utilForever — APPROVED** [review]
LGTM now. Thanks for your work! 🎉

File: bracket-geometry/src/rect.rs (hunk 3 above)
Comment (ChrisLee02) [reply bracket-geometry/src/rect.rs:240]: 저도 본 PR이 기능수정을 포함하는 이슈였다면, 최소한 points_set만을 노출시키고 for_each는 제거하는 방향이 좋다고 생각했구요, 궁극적으로는 1번대로 가는게 바람직해보입니다.
